integration-cli/docker_cli_by_digest_test.go

```diff
@@ -387,6 +387,33 @@ func (s *DockerRegistrySuite) TestListImagesWithDigests(c *check.C) {
 	}
 }
 
+func (s *DockerRegistrySuite) TestPsListContainersFilterAncestorImageByDigest(c *check.C) {
+	digest, err := setupImage(c)
+	c.Assert(err, check.IsNil, check.Commentf("error setting up image: %v", err))
+
+	imageReference := fmt.Sprintf("%s@%s", repoName, digest)
+
+	// pull from the registry using the <name>@<digest> reference
+	dockerCmd(c, "pull", imageReference)
+
+	// run a container based on that
+	out, _ := dockerCmd(c, "run", "-d", imageReference, "echo", "hello")
+	expectedID := strings.TrimSpace(out)
+
+	// Invalid imageReference
+	out, _ = dockerCmd(c, "ps", "-a", "-q", "--no-trunc", fmt.Sprintf("--filter=ancestor=busybox@%s", digest))
+	if strings.TrimSpace(out) != "" {
+		c.Fatalf("Expected filter container for %s ancestor filter to be empty, got %v", fmt.Sprintf("busybox@%s", digest), strings.TrimSpace(out))
+	}
+
+	// Valid imageReference
+	out, _ = dockerCmd(c, "ps", "-a", "-q", "--no-trunc", "--filter=ancestor="+imageReference)
+	if strings.TrimSpace(out) != expectedID {
+		c.Fatalf("Expected filter container for %s ancestor filter to be %v, got %v", imageReference, expectedID, strings.TrimSpace(out))
+	}
```

Comment:
In the other test (below) you check for ancestors and immediate parent by ID, name, etc... but not digest. Here you're just checking for immediate parent by digest.  I would hope checking for ancestor by digest would just work too, since its just another way to reference an image, but is it worth adding at least one check for "ancestor by digest" instead of just immediate parent?